src/compiler/checker.ts

```diff
@@ -268,6 +268,18 @@ namespace ts {
             },
             getJsxNamespace: () => unescapeLeadingUnderscores(getJsxNamespace()),
             getAccessibleSymbolChain,
+            nodeHasBadModifier: node => {
+                const sourceFile = getSourceFileOfNode(node);

```

Comment:
This is kind of ugly -- `checkGrammarModifiers` could be moved to a pure function returning an optional diagnostic but that have be a big diff.